runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/TestStreamTranslation.java

```diff
@@ -57,6 +62,48 @@
     return builder.build();
   }
 
+  private static TestStream<?> fromProto(
+      RunnerApi.TestStreamPayload testStreamPayload, RunnerApi.Components components)
+      throws IOException {
+
+    Coder<Object> coder =
```

Comment:
Yea, the reason I didn't do that is: where would `T` come from? It would be a generic parameter that was only present on the return type, which I avoid.

The practical difference is that callers of this method would be forced to do `TestStreamTranslation.<T>fromProto(...)` versus `(TestStream<T>) TestStreamTranslation.fromProto(...)`. Both are unchecked and unsafe (Java bummer) but the latter is a bit more explicit about the fact that a cast is going on.

Another option is to make a more clearly typed helper method. We do that elsewhere in the codebase but I just didn't go through the motion here.